api/clients/automation/automation_test.go

```diff
@@ -162,28 +187,34 @@ func TestAutomationClient_Update(t *testing.T) {
 	const payload = `{ "id" : "91cc8988-2223-404a-a3f5-5f1a839ecd45", "data" : "some-data1" }`
 	t.Run("Update  - try with adminAccess -if fails try without - OK", func(t *testing.T) {
 
-		responses := []testutils.ServerResponses{{
-			http.MethodPut: {
-				ResponseCode: http.StatusForbidden,
-				ResponseBody: payload,
-				ValidateRequestFunc: func(request *http.Request) {
-					adminAccessQP := request.URL.Query()["adminAccess"]
+		responses := []testutils.ResponseDef{
+			{
+				PUT: func(t *testing.T, req *http.Request) testutils.Response {
+					return testutils.Response{
+						ResponseCode: http.StatusForbidden,
+						ResponseBody: payload,
+					}
+				},
+				ValidateRequest: func(t *testing.T, req *http.Request) {
```

Comment:
nit/thought: With the responses now being functions that get the request as well as testing, I'm starting to wonder if having a dedicated Validate function is even required. I could just do the asserts as part of the response function - in the same way I would when writing a httptest.HandlerFunc.
What do you think?

Reply:
I left it separated, thinking further test utilities might provide some sort of "default" validators you could just plug in.